test/test_fields.jl

```diff
@@ -0,0 +1,3 @@
+function test_halos(input_field, array)
+    return isapprox(input_field.data, array, rtol = 1e-1)
```

Comment:
This seems **very large** tolerance. I'd expect machine precision accuracy. Why do we have to have 10% relative error tolerance here?

Reply:
Because the number of points were also less before. We can now have a smaller tolerance.

Reply:
I changed it to machine precision.

Reply:
But I disagree. It's not because the number of points was not high enough. This particular test should give machine prediction at any resolution.